### PR TITLE
Fixed Record Audio Waveform not visible #16258

### LINKED ISSUE #16258: [Emulator-only?] Record Audio Waveform not visible
image

It works fine in reviewer but in multimedia it has this issue now, Potrait -> Landscape -> Potrait, result in wave form to be invisible

_Originally posted by @criticalAY in https://github.com/ankidroid/Anki-Android/pull/16248#pullrequestreview-2016886844_

### PR DESCRIPTION
**Pull request by shalenMathew**
## Purpose / Description
fixing  #16258 the issue where audio wave disappears when orientation changes to landscape

## Fixes
* Fixes #16258 

## Approach

- issue occurred due Audio Waveform being kept GONE when config changed ,
solved the issue 

- audio form is now visible even in landscape mode

- changed the height of audio wave a bit , to suit in both landscape and potrait mode

## How Has This Been Tested?
- tested in emulator 

## Screenshot

https://github.com/ankidroid/Anki-Android/assets/119736953/4de4ae18-0ad7-4873-a996-e3c5110b9e5b

WhatsApp Image 2024-05-10 at 11 46 27 PM

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/layout/activity_audio_recording.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
         <com.ichi2.audio.AudioWaveform
             android:id="@+id/audio_waveform_view"
             android:layout_width="match_parent"
-            android:layout_height="300px"
+            android:layout_height="200px"
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/layout/activity_audio_recording.xml:45]: Could you test that with a height using `dp`? 200px may make sense for phones, but not on tablets

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

**Comment by david-allison** [comment]
Thanks!!

There was a reason that the audio waveform was hidden in landscape (I believe due to limited vertical space on phones) 

This doesn't address the issue that the flip from landscape to portrait does not reveal the control

**Comment by david-allison** [comment]
Forgot to ping, @criticalAY has context for the landscape change hiding the control

**Comment by criticalAY** [comment]
The audio wave form was kept hidden in landscape as the mobile screen don't have that much of space to accommodate both the waveform and the card content, when a user want to record while seeing the content of their card it is difficult if it's visible in landscape

**Review by criticalAY — CHANGES_REQUESTED** [review]
Thanks! but this is not the solution instead try fixing the original issue

**Comment by shalenMathew** [comment]
So whats the fix here ? Should i make any further changes ?

**Comment by shalenMathew** [comment]
Shall i change px to dp ?

**Comment by shalenMathew** [comment]
> You will need to fix the behavior mentioned in the issue which is emulator only 

Ok 👌

**Comment by david-allison** [comment]
@shalenMathew could you mark this as draft if it's not ready for review?

**Comment by shalenMathew** [comment]
> @shalenMathew could you mark this as draft if it's not ready for review?

I have changed the size of audio waveform to 100dp just notify me if i need to make any more changes

**Comment by criticalAY** [comment]
This is not what I meant, I will repeat what I said, you need to figure out what is the issue that causes the emulator bug and fix that instead of changing the height

**Comment by shalenMathew** [comment]
> This is not what I meant, I will repeat what I said, you need to figure out what is the issue that causes the emulator bug and fix that instead of changing the height

ok , i think it not a bug the waveform disappears coz the logic is set up that way , when the  orientation changes the waveform  visiblity is set to gone  , i think due to limited space of screen on landscape mode the logic is set up that way...

I think the logic works as intended ... Maybe its not an issue

**Comment by david-allison** [comment]
@shalenMathew first, you need to understand the problem: 

* We want the view hidden in landscape mode
* We don't want the view hidden in portrait mode
* If the user switches from portrait to landscape, then from landscape to portrait, the view is invisible in portrait, this is a bug 

----

The first step is to reproduce the problem in an emulator

Then explain what the bug is

Then fix the bug in a pull request

**Comment by shalenMathew** [comment]
> @shalenMathew first, you need to understand the problem:
> 
> * We want the view hidden in landscape mode
> * We don't want the view hidden in portrait mode
> * If the user switches from portrait to landscape, then from landscape to portrait, the view is invisible in portrait, this is a bug
> 
> The first step is to reproduce the problem in an emulator
> 
> Then explain what the bug is
> 
> Then fix the bug in a pull request

ok now i got it my bad, let me try to fix it

**Comment by criticalAY** [comment]
I am closing this PR as there are no changes here and i dont see any development recently, so the OP can figure out the right solution and get back.

**Pull request closed without merging**